plugins/chain_plugin/include/eosio/chain_plugin/tracked_votes.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <eosio/chain/types.hpp>
+#include <eosio/chain/block.hpp>
+#include <eosio/chain/controller.hpp>
+
+namespace eosio::chain_apis {
+   /**
+    * This class manages the ephemeral data that is needed by `get_finalizers_info` RPC call.
+    * There is no persistence and the cache is recreated when the class is instantiated
+    * based on the current state of the chain.
+    */
+   class tracked_votes {
+   public:
+
+      /**
+       * Instantiate a new tracked votes cache from the given chain controller
+       * The caller is expected to manage lifetimes such that this controller reference does not go stale
+       * for the life of the tracked votes cache
+       * @param chain - controller to read data from
+       */
+      tracked_votes( const class eosio::chain::controller& chain );
+      ~tracked_votes();
+
+      /**
+       * vote information for a given finalizer.
+       */
+      struct vote_info {
+         std::string          public_key;
+         std::string          description;
+         bool                 is_vote_strong{false};
+         uint32_t             voted_policy_generation{0};
+         chain::block_id_type voted_block_id;
+         uint32_t             voted_block_num{0};
+         fc::time_point       voted_block_timestamp;
```

Comment:
Do these identify the block which was voted on 
OR
Do these identify the block that contained the vote

Reply:
The block that was voted on.